src/cmd/cloud.js

```diff
@@ -779,11 +779,24 @@ module.exports = class CloudCommand extends CLICommandBase {
 	 */
 	_processDirIncludes(fileMapping, dirname, { followSymlinks } = {}){
 		dirname = path.resolve(dirname);
+		let files = new Set();
 
-		const includesFile = path.join(dirname, settings.dirIncludeFilename);
-		const ignoreFile = path.join(dirname, settings.dirExcludeFilename);
-		let hasIncludeFile = false;
+		this._getDefaultIncludes(files, dirname, { followSymlinks });
+		this._getCustomIncludes(files, dirname, { followSymlinks });
+		this._getDefaultIgnores(files, dirname, { followSymlinks });
+		this._getCustomIgnores(files, dirname, { followSymlinks });
 
+		// Add files to fileMapping
+		Array.from(files.values()).sort();
+		files.forEach((file) => {
+			// source relative to the base directory of the fileMapping (current directory)
+			const source = path.relative(fileMapping.basePath, file);
+			const target = path.relative(dirname, file);
+			fileMapping.map[target] = source;
+		});
+	}
+
+	_getDefaultIncludes(files, dirname, { followSymlinks }) {
 		// Recursively find source files
```

Comment:
Technically speaking, the default list should probably only have the following specific patterns that are found in src/ and lib/src only:

- project.properties
- src/*.[h|c](pp|xx)?
- src/*.ino
- src/build.mk
- lib/src/*.[h|c](pp|xx)?

We don't want to pick up anything in lib/usage/*.ino for example

Reply:
@eberseth Thanks for your feedback! I've made the suggested modifications. Just to confirm, are there any other significant patterns we should consider, given that we now focus on `src/*` and `lib/src/*` directories only, excluding other nested directories? With this, we will also no longer support the flat architecture like this - https://github.com/particle-iot/particle-cli/tree/master/test/__fixtures__/projects/legacy-flat

In summary:
1. Flat architecture will not be suppoorted - https://github.com/particle-iot/particle-cli/tree/master/test/__fixtures__/projects/legacy-flat
2. we have this structure of multiple header extensions in one of the tests - should we continue supporting it. https://github.com/particle-iot/particle-cli/tree/master/test/__fixtures__/projects/multiple-header-extensions
3. another one we support is this 'extended' structure where the helper is with in the /src https://github.com/particle-iot/particle-cli/tree/master/test/__fixtures__/projects/extended
4. What about symlinks like this? https://github.com/particle-iot/particle-cli/tree/master/test/__fixtures__/projects/symlink

A simple yes/no from you is sufficient for me to make the expected modifications.

Reply:
- No flat format is fine. 
- We never documented all of the other header file extensions but changing the list will likely break libraries in the future; I suspect that list happened when porting Arduino libraries. 
- The helper in src structure seems weird at first, but I suspect that is common for anyone who has divided their project into subdirectories, which is probably not that uncommon. 
- I didn't realize we supported symlinks that way but it could be useful when dealing with shared codebases. But you can also fix this with a build.mk, which might be a better way to do this than symlinks.

Reply:
Cool. Thanks for the inputs. @rickkas7.

Reply:
@eberseth To continue supporting certain legacy project structures, I will keep the changes to the original ones I had. To make changes to remove support for legacy formats, I am leaning towards getting the requirements sorted more clearly and will create a separate PR with those changes. For now, let's keep this PR focused on the particle.include and ignore files.